### PR TITLE
Remove deprecation warning caused by datadog.apm.enabled parameter

### LINKED ISSUE #461: Deprecated parameters used in example Helm charts
```
#################################################################
####               WARNING: Deprecation notice               ####
#################################################################

The option `datadog.apm.enabled` is deprecated, please use `datadog.apm.portEnabled` to enable TCP communication to the trace-agent.
The option `datadog.apm.socketEnabled` is enabled by default and can be used to rely on unix socket or name-pipe communication.
```

I'll submit a PR with the new configuration keys

### PR DESCRIPTION
**Pull request by christophetd**
#### Which issue this PR fixes

#461 

#### Special notes for your reviewer

Per https://github.com/DataDog/helm-charts/blob/main/charts/datadog/values.yaml#L261

#### Checklist

I don't believe it's needed to update the changelog / version for fixing an example

### CODE REVIEW

**Review by clamoriniere — COMMENTED** [review]
Hi @christophetd 

The changes looks good. 
But unfortunately even for updating docs or ci files, the chart version need to be bumped and so the CHANGELOG.md updated.

thanks
Cedirc

**Comment by christophetd** [comment]
Should be good now & rebased on latest `main` branch

**Comment by davidor** [comment]
@christophetd you need to run `.github/helm-docs.sh` to update the docs.

**Review by davidor — APPROVED** [review]

File: charts/datadog/README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Datadog
 
-![Version: 2.28.10](https://img.shields.io/badge/Version-2.28.10-informational?style=flat-square) ![AppVersion: 7](https://img.shields.io/badge/AppVersion-7-informational?style=flat-square)
+![Version: 2.28.11](https://img.shields.io/badge/Version-2.28.10-informational?style=flat-square) ![AppVersion: 7](https://img.shields.io/badge/AppVersion-7-informational?style=flat-square)
```
Comment (L3n41c) [inline charts/datadog/README.md:3]:
```suggestion
![Version: 2.28.11](https://img.shields.io/badge/Version-2.28.11-informational?style=flat-square) ![AppVersion: 7](https://img.shields.io/badge/AppVersion-7-informational?style=flat-square)
```

**Review by L3n41c — APPROVED** [review]

**Pull request merged**